pkg/common/common.go

```diff
@@ -216,3 +217,20 @@ func StripDefaultPorts(fromUrl string) (string, error) {
 	}
 	return r.URL.String(), nil
 }
+
+// GetImagePullPolicy get imagePullPolicy for a container, based on its image.
+// If image contains a sha256 or sha512 digest, use IfNotPresent; otherwise, Always.
+// Reference: https://github.com/distribution/distribution/blob/v2.7.1/reference/reference.go
+func GetImagePullPolicy(image string) corev1.PullPolicy {
+	matched, err := regexp.Match("@sha256:[a-f0-9]{64}", []byte(image))
+	if err == nil && matched {
```

Comment:
should we at least log if an error occurs?

Reply:
Sure.